lib/host/wasi_nn/wasinnfunc.cpp

```diff
@@ -6,46 +6,434 @@
+                  "to build it.");
+#endif
+  } else {
+    logger->error("Current backend is not supported.");
+  }
+  return -1;
 }
 
 Expect<uint32_t> WasiNNGetOuput::body(
-    Runtime::Instance::MemoryInstance *MemInst [[maybe_unused]],
-    uint32_t Context [[maybe_unused]], uint32_t Index [[maybe_unused]],
-    uint32_t OutBuffer [[maybe_unused]],
-    uint32_t OutBufferMaxSize [[maybe_unused]],
-    uint32_t BytesWrittenPtr [[maybe_unused]]) {
-  return 0;
+    Runtime::Instance::MemoryInstance *MemInst, uint32_t Context,
+    uint32_t Index [[maybe_unused]], uint32_t OutBuffer,
+    uint32_t OutBufferMaxSize [[maybe_unused]], uint32_t BytesWrittenPtr) {
+  if (MemInst == nullptr) {
+    return Unexpect(ErrCode::ExecutionFailed);
+  }
+
+  auto logger = spdlog::get("wasi-nn");
+  [[maybe_unused]] uint8_t *OutBufferPtr;
+  [[maybe_unused]] uint32_t *BytesWritten;
+
+  if (Ctx.GraphContextBackends.size() <= Context) {
+    logger->error("get_output: Execution Context does not exist");
+    return -1;
+  }
+
+  OutBufferPtr = MemInst->getPointer<uint8_t *>(OutBuffer, 1);
+  BytesWritten = MemInst->getPointer<uint32_t *>(BytesWrittenPtr, 1);
+
+  if (Ctx.GraphContextBackends[Context] == Ctx.BackendsMapping.at("OpenVINO")) {
+#ifdef WASMEDGE_WASINN_BUILD_OPENVINO
+    IEStatusCode Status;
+    OpenVINOSession *Session = Ctx.OpenVINOInfers[Context];
+    ie_network_t *Network = Session->network;
+    ie_infer_request_t *InferRequest = Session->infer_request;
+    ie_blob_t *OutputBlob = nullptr;
+    ie_blob_buffer_t BlobCBuffer;
+    char *OutputName = nullptr;
+    int BlobSize;
+
+    Status = ie_network_get_output_name(Network, Index, &OutputName);
+    if (Status != OK) {
+      logger->error("Unable to find output name correctly with Index:{}",
+                    Index);
+      return -1;
+    }
+    Status = ie_network_set_output_precision(Network, OutputName, FP32);
+    if (Status != OK) {
+      logger->error("Unable to set output precision correctly with Index:{}",
+                    Index);
+      return -1;
+    }
+
+    Status = ie_infer_request_get_blob(InferRequest, OutputName, &OutputBlob);
+    if (Status != OK || OutputName == nullptr) {
+      logger->error("Unable to retrieve output tensor correctly", Index);
+      return -1;
+    }
+    Status = ie_blob_size(OutputBlob, &BlobSize);
+    Status = ie_blob_get_cbuffer(OutputBlob, &BlobCBuffer);
+    if (Status != OK) {
+      logger->error("Unable to retrieve output tensor correctly", Index);
+      return -1;
+    }
+    float *BlobData = (float *)(BlobCBuffer.cbuffer);
+    size_t BytesToWrite = BlobSize * 4;
+    if (BytesToWrite > OutBufferMaxSize)
+      BytesToWrite = OutBufferMaxSize;
+    uint8_t *CastedOutputData = reinterpret_cast<uint8_t *>(BlobData);
+
+    for (size_t I = 0; I < BytesToWrite; I++) {
+      OutBufferPtr[I] = CastedOutputData[I];
+    }
+    *BytesWritten = BytesToWrite;
+    Ctx.OpenVINOOutputs.push_back(OutputBlob);
```

Comment:
It seems like the output data is copied into the `OutBufferPtr`.
Why we should store the `OutputBlob`?

Reply:
Cause I'm not sure whether the `ie_blob_free` will free the tensor inside the `InferRequest`, users can use the same `Context` id to access this output which may cause an error. 
I have traced down the source code and still can't make sure.